src/Template/Bake/config/snapshot.ctp

```diff
@@ -25,50 +26,87 @@ class <%= $name %> extends AbstractMigration
-            $constraintColumns = $constraint['columns'];
-            sort($constraintColumns);
-            if ($constraint['type'] !== 'unique'):
-                $foreignKeys += $constraint['columns'];
+        <%- endforeach;
+            $tableConstraints = $this->Migration->constraints($table);
+            if (!empty($tableConstraints)):
+                $constraints[$table] = $tableConstraints;
+
+                foreach ($constraints[$table] as $constraint):
+                    if ($constraint['type'] !== 'unique'):
+                        $foreignKeys += $constraint['columns'];
+                    endif;
+                    if ($constraint['columns'] !== $primaryKeysColumns): %>
+            ->addIndex(
+                [<% echo $this->Migration->stringifyList($constraint['columns'], ['indent' => 5]); %>]<% echo ($constraint['type'] === 'unique') ? ',' : ''; %>
 
-                $columnsList = '\'' . $constraint['columns'][0] . '\'';
-                if (count($constraint['columns']) > 1):
-                    $columnsList = '[' . $this->Migration->stringifyList($constraint['columns'], ['indent' => 5]) . ']';
-                endif;
+                <%- if ($constraint['type'] === 'unique'): %>
+                ['unique' => true]
+                <%- endif; %>
+            )
+                    <%- endif;
+                endforeach;
+            endif;
 
-                if (is_array($constraint['references'][1])):
-                    $columnsReference = '[' . $this->Migration->stringifyList($constraint['references'][1], ['indent' => 5]) . ']';
-                else:
-                    $columnsReference = '\'' . $constraint['references'][1] . '\'';
-                endif; %>
+            foreach($this->Migration->indexes($table) as $index):
+                sort($foreignKeys);
+                $indexColumns = $index['columns'];
+                sort($indexColumns);
+                if ($foreignKeys !== $indexColumns):
+                %>
+                ->addIndex(
+                    [<%
+                        echo $this->Migration->stringifyList($index['columns'], ['indent' => 5]);
+                    %>]
+                )
+            <%- endif;
+            endforeach; %>
+            -><%= $tableMethod %>();
+
+    <%- endforeach; %>
+    <%- foreach ($constraints as $table => $tableConstraints):
+            foreach ($tableConstraints as $constraint):
+                $constraintColumns = $constraint['columns'];
+                sort($constraintColumns);
+                if ($constraint['type'] !== 'unique'):
+                    $columnsList = '\'' . $constraint['columns'][0] . '\'';
+                    if (count($constraint['columns']) > 1):
+                        $columnsList = '[' . $this->Migration->stringifyList($constraint['columns'], ['indent' => 5]) . ']';
+                    endif;
+                    $dropForeignKeys[$table] = $columnsList;
```

Comment:
Should be appended. In case on multiple FK constraints, this will only store the last one and will be causing trouble for the ``down()`` method